### PR TITLE
Users/mitokic/pca feature

### PR DESCRIPTION
**Pull request by mitokic**
Added pca feature to run PCA as recipe step. 

By default, will run PCA for day/week date types, and will run PCA on global models regardless of date type. 

Main benefit is to speed up model run time with default inputs, and decreasing chances of memory/RAM issues when running forecasts in parallel (especially for day/week date types).

### CODE REVIEW

File: R/configure_forecast_run.R [hunk 1]
```diff
@@ -49,9 +49,7 @@ get_lag_periods <- function(lag_periods,
                     "quarter" = c(1,2,3,4),
                     "month" =  c(1, 2, 3, 6, 9, 12),
                     "week" = c(1, 2, 3, 4, 8, 12, 24, 48, 52),
-                    "day" = c(1, 2, 3, 4, 5, 6, 7, 14, 
-                              21, 28, 28*2, 28*3, 28*6, 
-                              28*9, 28*12, 365)
+                    "day" = c(7, 14, 21, 28, 60, 90, 180, 365)
```
Comment (AKannanMSFT) [inline R/configure_forecast_run.R:52]: I am in general concerned about #77 - Do we want to address this as a part of this PR ?

**Review by AKannanMSFT — APPROVED** [review]
Please treat this as a conditional Approval - if you don't want to address #77, this PR is approved

**Review by aadharshkannan — APPROVED** [review]
Merge change

**Review by AKannanMSFT — APPROVED** [review]
Merge changes. Approved

File: R/configure_forecast_run.R (hunk 1 above)
Comment (mitokic) [reply R/configure_forecast_run.R:52]: Great call out. My comments are in the issue #77

Comment (mitokic) [reply R/configure_forecast_run.R:52]: I propose tackling this in another PR

**Pull request merged**